From 67850e5d16513a62bc552834acabff9907db71a6 Mon Sep 17 00:00:00 2001
From: pkandiX <pradeep.kumarx.kandi@intel.com>
Date: Tue, 8 Jan 2013 15:52:01 +0530
Subject: Game crashes when played with OGG audio

BZ: 76736

Checking for duration if it is greater than zero
and then sending END_OF_STREAM error if required.

Category: aosp improvement
Domain: AUDIO_MEDIA-Audio Framework
Origin: Internal
Upstream-Candidate: yes

Change-Id: I6d82000953dc576abcc2d1c8f5fa0eed1f3ed32d
Orig-Change-Id: I84599666bed07bdb1e5d6f5cdcc1613be70edd1b
Signed-off-by: pkandiX <pradeep.kumarx.kandi@intel.com>
---
 media/libstagefright/OggExtractor.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/OggExtractor.cpp b/media/libstagefright/OggExtractor.cpp
index 2cafa3a..8ddf120 100644
--- a/media/libstagefright/OggExtractor.cpp
+++ b/media/libstagefright/OggExtractor.cpp
@@ -507,7 +507,7 @@ status_t MyVorbisExtractor::readNextPacket(MediaBuffer **out) {
             if (timeUs > 0) {
                 int64_t durationUs = 0;
                 if (mMeta->findInt64(kKeyDuration, &durationUs)) {
-                    if (durationUs < timeUs) {
+                    if (durationUs < timeUs && durationUs > 0) {
                         LOGV("Reached End of File");
                         return ERROR_END_OF_STREAM;
                     }
-- 
1.7.4.1

